vendored_yarn not defining dependency properly causing Bazel to not include the linked yarn in the external folder

# 🐞 bug report

### Affected Rule

vendored_yarn inside node_repositories

<pre><code>
node_repositories(
    node_version = "10.12.0",
    vendored_node = "@vendored_node_10_12_0//:node-v10.12.0-linux-x64",
    vendored_yarn = "@vendored_yarn_1_10_0//:yarn-v1.10.0",
)
</code></pre>


### Description

When trying to use a vendored version of yarn via vendored_yarn bazel doesn't include the specified folder inside the external folder of the output user root. We have tested both using a local_repository and the http_archive like the `vendored_node_and_yarn` example and both fail to include yarn which causes yarn commands to not work (eg: `bazel run @nodejs//:yarn_node_repositories`). Doing a `bazel sync` seems to force bazel to detect the dependency and fixes the issue, but it is not a great fix to ask every one using our repo to do it locally.


## 🔬 Minimal Reproduction

Using the basic workspace setup at https://github.com/ceraetes/rules_nodejs_vendored_yarn (modified version of the example to use self-managed dependencies) and trying to run `bazel run @nodejs//:yarn_node_repositories` will trigger the issue.

## 🔥 Exception or Error

<pre><code>

Running yarn  in rules_nodejs_vendored_yarn
internal/modules/cjs/loader.js:582
    throw err;
    ^

Error: Cannot find module 'bazel_output\wk6kt6hu\external\vendored_yarn_1_10_0\yarn-v1.10.0\bin\yarn.js'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:580:15)
    at Function.Module._load (internal/modules/cjs/loader.js:506:25)
    at Function.Module.runMain (internal/modules/cjs/loader.js:741:12)
    at startup (internal/bootstrap/node.js:285:19)
    at bootstrapNodeJSCore (internal/bootstrap/node.js:739:3)

</code></pre>


## 🌍  Your Environment

**Operating System:**

<pre>
  <code>
    Windows 10
  </code>
</pre>

**Output of `bazel version`:**

<pre>
  <code>
    Build label: 3.3.1
    Build target: bazel-out/x64_windows-opt/bin/src/main/java/com/google/devtools/build/lib/bazel/BazelServer_deploy.jar
    Build time: Tue Jun 30 15:13:25 2020 (1593530005)
    Build timestamp: 1593530005
    Build timestamp as int: 1593530005
  </code>
</pre>

**Rules_nodejs version:**

(Please check that you have matching versions between WORKSPACE file and `@bazel/*` npm packages.)

<pre>
  <code>
     2.0.0-rc.1
  </code>
</pre>

**Anything else relevant?**

--- a/internal/node/node_repositories.bzl
+++ b/internal/node/node_repositories.bzl
@@ -422,6 +422,13 @@ def _prepare_node(repository_ctx):
     yarn_bin_label = ("%s/bin/yarn.js" % yarn_package) if not is_windows else ("%s/yarn.cmd" % yarn_package)
     yarn_script = "%s/bin/yarn.js" % yarn_path
 
+    # Ensure that the "vendored" binaries are resolved
+    # Just requesting their path from the repository context is enough to eager-load them
+    if repository_ctx.attr.vendored_node:
+        repository_ctx.path(Label(node_bin_label))
+    if repository_ctx.attr.vendored_yarn:
+        repository_ctx.path(Label(yarn_bin_label))
+
     entry_ext = ".cmd" if is_windows else ""
     node_entry = "bin/node%s" % entry_ext
     npm_entry = "bin/npm%s" % entry_ext

